lms/djangoapps/progress/tests.py

```diff
@@ -141,6 +147,31 @@ def test_save_completion(self):
         )
         self.assertIsNotNone(completion_fetch)
 
+    def test_check_notifications(self):
+        """
+        Save a CourseModuleCompletion and fetch it again
+        """
+        module = self.get_module_for_user(self.user, self.course, self.problem4)
+        module.system.publish(module, 'progress', {})
+
+        completion_fetch = CourseModuleCompletion.objects.get(
+            user=self.user.id,
+            course_id=self.course.id,
+            content_id=self.problem4.location
+        )
+        self.assertIsNotNone(completion_fetch)
+
+        # since we are alone, then we should be listed as first
+        self.assertEqual(get_notifications_count_for_user(self.user.id), 1)
+
+        # progressing on a 2nd item, shouldn't change our positions, because
+        # we're the only one in this course
+        module = self.get_module_for_user(self.user, self.course, self.problem5)
+        module.system.publish(module, 'progress', {})
+
+        # since we are alone, then we should be listed as first
+        self.assertEqual(get_notifications_count_for_user(self.user.id), 1)
+
```

Comment:
I'd like to see a self.assertNumQueries check here to ensure we're hitting the database only as much as we expect.